### PR TITLE
6848 retire gov data

### LINKED ISSUE #6848: Retire {Get/List}GovernanceData
As of now the grpc API provide an endpoint to get a list of votes per party and proposal, and an hybrid endpoint {Get/List}GovernanceData which takes a bunch of filter and returns all machine proposals with their associated votes.
While this sounds nice on paper, this becomes a mess with management of the pagination of votes etc.

Let's remove this endpoint and add a simple {Get/List}Proposal, with the same filters but returning only the propsals.

### PR DESCRIPTION
**Pull request by EVODelavega**
Closes #6848

### CODE REVIEW

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 1]
```diff
@@ -285,11 +285,17 @@ service TradingDataService {
...
 
+    // Get a single proposal
+    rpc GetProposal(GetProposalRequest) returns (GetProposalResponse);
+
     // Governance list
     //
     // List proposals using a cursor based pagination model
     rpc ListGovernanceData(ListGovernanceDataRequest) returns (ListGovernanceDataResponse);
 
+    // List proposals using a cursor based pagination model
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:296]: What's the difference between this and ListGovernanceData(ListGovernanceDataRequest)? 

The description is the same for both, does one need an update?

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 2]
```diff
@@ -1732,6 +1747,55 @@ message ListGovernanceDataRequest {
...
+message ListProposalsRequest {
+    enum Type {
+        TYPE_UNSPECIFIED = 0;
+        // List all proposals
+        TYPE_ALL = 1;
+        // List new market proposals
+        TYPE_NEW_MARKET = 2;
+        // List update market proposals
+        TYPE_UPDATE_MARKET = 3;
+        // List change Vega network parameter proposals
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1759]:
```suggestion
        // List proposals for changing Vega network parameters
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 3]
```diff
@@ -1732,6 +1747,55 @@ message ListGovernanceDataRequest {
...
+        TYPE_NETWORK_PARAMETERS = 4;
+        // New asset proposals
+        TYPE_NEW_ASSET = 5;
+        // Proposals for creating a new free form proposal
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1763]:
```suggestion
        // List free form proposals for community requests
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -1732,6 +1747,55 @@ message ListGovernanceDataRequest {
...
+        TYPE_UNSPECIFIED = 0;
+        // List all proposals
+        TYPE_ALL = 1;
+        // List new market proposals
+        TYPE_NEW_MARKET = 2;
+        // List update market proposals
+        TYPE_UPDATE_MARKET = 3;
+        // List change Vega network parameter proposals
+        TYPE_NETWORK_PARAMETERS = 4;
+        // New asset proposals
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1761]:
```suggestion
        // List new asset proposals
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 5]
```diff
@@ -1732,6 +1747,55 @@ message ListGovernanceDataRequest {
...
+        TYPE_NEW_ASSET = 5;
+        // Proposals for creating a new free form proposal
+        TYPE_NEW_FREE_FORM = 6;
+        // Update asset proposals
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1765]:
```suggestion
        // List asset update proposals
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 6]
```diff
@@ -1732,6 +1747,55 @@ message ListGovernanceDataRequest {
...
 }
 
+message ListProposalsRequest {
+    enum Type {
+        TYPE_UNSPECIFIED = 0;
+        // List all proposals
+        TYPE_ALL = 1;
+        // List new market proposals
+        TYPE_NEW_MARKET = 2;
+        // List update market proposals
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1757]:
```suggestion
        // List market update proposals
```

**Comment by jeremyletang** [comment]
closing for now as it's massively out of date, and would be a difficult to manage breaking change now. let's discuss that again in the future.

**Pull request closed without merging**